### DISCUSSION TITLE
flux does not delete unmanaged namespace in the cluster

### DISCUSSION THREAD

**Discussion by himani-dangwal** [fluxcd/flux2#2134] [Q&A] [answered]
We are using Flux2 and helm3 for our deployment. Previously when we deleted the kustomization files from the Git repo then after the sync the namespace and its corresponding objects started getting delete from the cluster. But recently it has stopped working. 
 Even if we delete the files from the git repo the namespace and its objects are there in the cluster . If we have to delete them, we manually delete the helmrelease files from the cluster.  
  Is there any change in the implementation recently or we are missing something ?

**Comment by kingdonb** [comment, marked as answer]
Sorry for the trouble, and that nobody has gotten back to you.

It should be normal that the resources Flux manages will be deleted if you delete them from the Git repository and the Flux Kustomization had `spec.prune` set to `true`.

It is possible that you've deleted the directory that Kustomization was syncing from, in that case I think Flux will not delete everything in case it is an error such as, Kustomization has been pointed at an invalid path.

It is also possible that you ran into a bug which has been fixed. The current Flux release is 0.32.x and has many advancements since this time. Please let us know if you are still working on Flux, if there's anything we can help with currently!

I'm going to close this because it has aged out, but welcome your response if you need anything further.
